modules/core/04-channel/keeper/upgrade.go

```diff
@@ -22,115 +14,64 @@ func (k Keeper) ChanUpgradeInit(
 	ctx sdk.Context,
 	portID string,
 	channelID string,
-	chanCap *capabilitytypes.Capability,
-	proposedUpgradeChannel types.Channel,
-	counterpartyTimeoutHeight clienttypes.Height,
-	counterpartyTimeoutTimestamp uint64,
-) (upgradeSequence uint64, previousVersion string, err error) {
+	upgradeFields types.UpgradeFields,
+	upgradeTimeout types.UpgradeTimeout,
+) (proposedUpgrade types.Upgrade, err error) {
```

Comment:
I wonder if the named return values are still necessary?

Reply:
I'm not a huge fan of named return arguments anyway (magic var initialisation ) and now with fewer return args I would agree